woocommerce/woocommerce-ios
# Orders List: Improving Error Handling

### Details:
In this PR we're:

- Removing the Orders List "Error Overlay" (which was being displayed whenever Sync failed)
- Implementing a super cool Notice, which will show up on top of the tabBar, and will let users retry the Sync OP.

cc @bummytime @mindgraffiti (Thanks in advance!!!)
Closes #286

---

### Scenario: Placeholder
1. Fresh install the app
2. Log into your Woo Store
3. Open the Orders tab
4. Verify the Skeleton shows up while the orders are loaded.


### Scenario: Empty / Filtered
1. Check out the latest commit of this PR
2. Log into your Woo Store
3. Open the Orders tab
4. Filter by Order Status (and select a status that has no visible orders)
5. Verify the Empty State / Filtered UI shows up
6. Press the `Remove Filter` button
7. Verify the unfiltered results show up


### Scenario: Empty State / Unfiltered
1. Check out this commit: [ff3772d](
https://github.com/woocommerce/woocommerce-ios/commit/ff3772d5469bc0c3a7cbd1ce4938c28e8dc4f17b)
2. Log into your Woo Store
3. Open the Orders tab
4. Verify that the Empty State UI shows up. 
5. Press the Action Button
6. Verify that your Store shows up in a WebView

### Scenario: Error Notice
1. Check out this commit: [3546d24](
https://github.com/woocommerce/woocommerce-ios/commit/3546d24a8c013bb49b01ce934ad206f19bf79952)
2. Log into your Woo Store
3. Open the Orders tab
4. Verify the Error State is Onscreen
5. Press the `Retry` button
6. Verify shortly after the error screen shows up again
